packages/cli/src/index.ts

```diff
@@ -12,183 +12,285 @@ import {
+    handleAsyncOperationErrors(
+      watch(defaultOptions),
+      'error during watch project'
+    );
+  }
+});
+
+const previewCommand = defineCommand({
+  meta: {
+    name: 'preview',
+    description: 'compile the project in watch mode'
+  },
+  args: {
+    root: {
+      type: 'positional',
+      description: 'root path',
+      required: false,
+      valueHint: 'path'
+    },
+    port: { type: 'string', description: 'specify port' },
+    open: {
+      type: 'boolean',
+      description: 'open browser on server preview start'
     }
-  );
-
-cli
-  .command('clean [path]', 'Clean up the cache built incrementally')
-  .option(
-    '--recursive',
-    'Recursively search for node_modules directories and clean them'
-  )
-  .action(async (rootPath: string, options: ICleanOptions) => {
-    const { root } = resolveCliConfig(rootPath, options);
-    const { clean } = await resolveCore();
+  },
+  async run({ args }: { args: FarmCLICommonOptions & FarmCLIPreviewOptions }) {
+    const { root, configPath } = resolveCliConfig(args.root, args.config);
 
+    const resolvedOptions = resolveCommandOptions(args as GlobalFarmCLIOptions);
+    const defaultOptions = {
+      root,
+      mode: args.mode,
+      server: resolvedOptions,
+      configPath,
+      port: resolvedOptions.port
+    };
+
+    const { preview } = await resolveCore();
+    handleAsyncOperationErrors(
+      preview(defaultOptions),
+      'Failed to start preview server'
+    );
+  }
+});
+
+const cleanCommand = defineCommand({
+  meta: {
+    name: 'clean',
+    description: 'Clean up the cache built incrementally'
+  },
+  args: {
+    root: {
+      type: 'positional',
+      description: 'root path',
+      required: false,
+      valueHint: 'path'
+    },
+    recursive: {
+      type: 'boolean',
+      alias: 'r',
+      description:
+        'Recursively search for node_modules directories and clean them'
+    }
+  },
+  async run({ args }: { args: FarmCLICommonOptions & ICleanOptions }) {
+    const { root } = resolveCliConfig(args.root, args.config);
+
+    const { clean } = await resolveCore();
     try {
-      await clean(root, options?.recursive);
+      await clean(root, args.recursive);
     } catch (e) {
       const { Logger } = await import('@farmfe/core');
       const logger = new Logger();
       logger.error(`Failed to clean cache: \n ${e.stack}`);
       process.exit(1);
     }
-  });
-
-// Listening for unknown command
-cli.on('command:*', async () => {
-  const { Logger } = await import('@farmfe/core');
-  const logger = new Logger();
-  logger.error(
-    'Unknown command place Run "farm --help" to see available commands'
-  );
+  }
+});
+
+const main = defineCommand({
+  meta: {
+    name: 'farm',
+    version
+  },
+  args: {
+    config: {
+      type: 'string',
+      alias: 'c',
+      description: 'use specified config file'
+    },
+    mode: { type: 'string', alias: 'm', description: 'set env mode' },
```

Comment:
shot name it doesnt work

Reply:
I'll check this later, but I truely dealed with this issue with this funciton: https://github.com/farm-fe/farm/pull/1517/files#diff-852d90245bbe147114049b7940720369a5dd1c564e5d1ab29b1bf8eae193cef3R192-R207